src/test/java/tests/NullSpecTest.java

```diff
@@ -117,10 +117,8 @@ public TypecheckResult adjustTypecheckResult(TypecheckResult testResult) {
 
     for (ListIterator<TestDiagnostic> i = unexpected.listIterator(); i.hasNext(); ) {
       TestDiagnostic diagnostic = i.next();
-      DetailMessage detailMessage = DetailMessage.parse(diagnostic, null);
-      if (detailMessage.hasDetails()) {
-        // Replace diagnostics that can be parsed with DetailMessage diagnostics.
-        i.set(detailMessage);
+      if (diagnostic instanceof DetailedTestDiagnostic) {
```

Comment:
As the normal tests are executed without `-Adetailedmsgtext`, this is likely a dead branch. I'll address this in a separate PR.